src/pages/SampleListPage.tsx

```diff
@@ -53,16 +56,41 @@ export const SampleListPage = () => {
     }));
   }, [data]);
 
+  useEffect(() => {
```

Comment:
Would it be better to use `useMemo()` instead?

Reply:
Do you think so? It handles the user selection, i. e. something that would change quite often, isn't useMemo() preferable rather for data which is rarely changed? But I could change it to useMemo(), if you think that is is a better option.